spec/unit/Java.spec.js

```diff
@@ -0,0 +1,156 @@
+        });
+
+        it('with JAVA_HOME / without javac', async () => {
+            spyOn(utils, 'forgivingWhichSync').and.returnValue('');
+            process.env.JAVA_HOME = '/tmp/jdk';
+
+            await Java._ensure();
+
+            expect(process.env.PATH).toMatch(/\/tmp\/jdk+(\/|\\)bin/);
+        });
+
+        it('without JAVA_HOME / with javac - Mac OS X - success', async () => {
+            spyOn(utils, 'forgivingWhichSync').and.returnValue('/tmp/jdk/bin');
+            process.env.JAVA_HOME = '';
+            const fsSpy = jasmine.createSpy('fs').and.returnValue(true);
+            Java.__set__('fs', {
+                existsSync: fsSpy
+            });
+            Java.__set__('execa', jasmine.createSpy('execa').and.returnValue(Promise.resolve({
+                stdout: '/tmp/jdk'
+            })));
+
+            await Java._ensure();
+
+            expect(fsSpy).toHaveBeenCalledWith('/usr/libexec/java_home');
+            expect(process.env.JAVA_HOME).toBe('/tmp/jdk');
+        });
+
+        it('without JAVA_HOME / with javac - Mac OS X - error', async () => {
+            spyOn(utils, 'forgivingWhichSync').and.returnValue('/tmp/jdk/bin');
+            process.env.JAVA_HOME = '';
+            Java.__set__('fs', {
+                existsSync: jasmine.createSpy('fs').and.returnValue(true)
+            });
+            Java.__set__('execa', jasmine.createSpy('execa').and.returnValue(Promise.reject({
+                shortMessage: 'test error'
+            })));
+            const emitSpy = jasmine.createSpy('events.emit');
+            Java.__set__('events', {
+                emit: emitSpy
+            });
+
+            await expectAsync(Java._ensure()).toBeRejectedWithError(CordovaError, /Failed to find 'JAVA_HOME' environment variable/);
+
+            expect(emitSpy).toHaveBeenCalledWith('verbose', 'test error');
+        });
+
+        it('derive from javac location - success', async () => {
+            spyOn(utils, 'forgivingWhichSync').and.returnValue('/tmp/jdk/bin');
+            process.env.JAVA_HOME = '';
+            Java.__set__('fs', {
+                existsSync: jasmine.createSpy('fs').and.callFake((path) => {
+                    if (/java_home$/.test(path)) {
+                        return false;
+                    } else {
+                        return true;
+                    }
+                })
+            });
+
+            await Java._ensure();
+
+            expect(process.env.JAVA_HOME).toBe('/tmp');
+        });
+
+        it('derive from javac location - error', async () => {
+            spyOn(utils, 'forgivingWhichSync').and.returnValue('/tmp/jdk/bin');
+            process.env.JAVA_HOME = '';
+            Java.__set__('fs', {
+                existsSync: jasmine.createSpy('fs').and.callFake((path) => {
+                    if (/java_home$/.test(path)) {
+                        return false;
+                    } else {
+                        return false;
+                    }
```

Comment:
Always returns `false`? Seems like the `if` condition is not needed.

Reply:
Yes, this seems like something I missed. I'll re investigate probably on the weekend to determine what this should be. Would have thought I had picked up on this when writing the tests...